toolkit/tools/imagegen/configuration/user.go

```diff
@@ -74,13 +75,13 @@ func (p *User) UIDIsValid() (err error) {
 		uidLowerBound = 0 // root user
 		uidUpperBound = 60000
 	)
-	if p.UID != "" {
+	if strings.TrimSpace(p.UID) != "" {
 		uidNum, err := strconv.Atoi(p.UID)
 		if err != nil {
-			return fmt.Errorf("Failed to convert UID (%s) to a number", p.UID)
+			return fmt.Errorf("failed to convert UID (%s) to a number", p.UID)
 		}
 		if uidNum < uidLowerBound || uidNum > uidUpperBound {
-			return fmt.Errorf("invalid value for UID (%s)", p.UID)
+			return fmt.Errorf("invalid value for UID (%s), not within [0, 60000]", p.UID)
```

Comment:
I'd suggest using the constants instead of hard-coded strings so we always get the right numbers and make a change in only one place in the (unlikely) event that these bounds ever change. But that's good practice in general. Since you're using these bounds in the tests as well, you could move the constants to be global and re-use them inside the tests. As long as the code and its tests are inside the same package (they are, that's usually the case anyway) private globals will be visible.

Same comment for `upperBoundChage` and `noExpiration` below.

Reply:
I agree that here we should use the constants and not hard code